addons/koji/common/src/main/java/org/commonjava/indy/koji/content/KojiContentManagerDecorator.java

```diff
@@ -299,6 +286,35 @@ public Transfer getTransfer( final ArtifactStore store, final String path, final
         return defValue;
     }
 
+    private boolean isKojiPullingEnabled( final Group store, final String path )
+    {
+        if ( !config.getEnabled() )
+        {
+            logger.debug( "Koji content-manager decorator is disabled." );
+            logger.debug( "When koji addon is disabled , path:{},config instance is {}", path, config.toString() );
+            return false;
+        }
+
+        if ( !config.isEnabledFor( store.getName() ) )
```

Comment:
Actually this part should be either dropped or made an alternative to the new koji-pull switch in metadata. It is the part deciding if brew pull is active for a group based on the patterns in config, right? That was meant to be replaced by the new metadata key approach. Keeping both in place does not make sense IMO. Or it would for transition period, but not require them both to match the condition, but make them alternative to each other - either a build group name matches a pattern in config or the metadata key is present with value "true". I'll add a comment to the jira to discuss it there.

Reply:
@pkocandr I've fixed the logic as the description said now. Does that make sense?